Python/chapter02/2.2 - Return Kth to Last/miguel_2.2_soln.py

```diff
@@ -84,79 +87,39 @@ def kth_to_last(ll: LinkedList, k: int) -> Node:
     Going to reverse the linked list and then
     count k steps.
     Runtime: O(N)
-    Space Complexity: O(N)
+    Space Complexity: O(1)
     :param ll: a linked list
-    :param k: an integer where k >= 0
+    :param k:
+        an integer where k > 0 and k < ll size
     :return:
         kth to last Node from the linked list
         or None
     """
-    if k <= 0:
-        return None
-    # build reversed linked list
-    reversed_ll = LinkedList(ll.head.data)
-    n = ll.head.next
-    while n is not None:
-        reversed_ll.append_to_head(n.data)
-        n = n.next
-    # go k steps
+    if k <= 0 or k > ll.size:
+        raise IndexError('list index out of range')
+    # go size - k steps
+    n = ll.head
     i = 1
-    n = reversed_ll.head
-    while n is not None and i < k:
+    while n is not None and i <= ll.size - k:
```

Comment:
maybe use a `for` instead?

Reply:
updated to use for loop

Reply:
could you re-approve the PRs? My nit commits changed status from approved to unreviewed.

I understand that you still can't merge them due to privilege reasons, though.